### ISSUE TITLE
How to train my own dataset

### ISSUE THREAD

**Issue by InstantWindy** [ayooshkathuria/pytorch-yolo-v3#93]
Hi！ If I want to tain my own data with your model, how to run it? Thanks!

**Comment by glenn-jocher** [comment]
https://github.com/ultralytics/yolov3/wiki/Train-Custom-Data

**Comment by juanmed** [comment]
> https://github.com/ultralytics/yolov3/wiki/Train-Custom-Data

Does the train.py file in the repository you link to works for with the darknet.py file in this (ayooshkathuria/pytorch-yolo-v3) repository?

**Comment by emmbertelen** [comment]
> 
> 
> > https://github.com/ultralytics/yolov3/wiki/Train-Custom-Data
> 
> Does the train.py file in the repository you link to works for with the darknet.py file in this (ayooshkathuria/pytorch-yolo-v3) repository?



Hey, did u find it out?

**Comment by glenn-jocher** [comment]
The YOLOv3 repository below is MacOS/Windows/Linux compatible, includes multigpu/multiscale, performs rectangular+batchnorm fused inference on images, videos, webcams, and even an iOS app. It tests slightly better than darknet (**60.7 COCO2014 mAP** with `yolov3-spp.weights`), and trains FP16/FP32 custom datasets from scratch to darknet performance, all using PyTorch :)
https://github.com/ultralytics/yolov3

<table style="width:100%">
  <tr>
    <td>
      <img src="https://user-images.githubusercontent.com/26833433/61591130-f7beea00-abc2-11e9-9dc0-d6abcf41d713.jpg">
    </td>
    <td align="center">
    <a href="https://www.ultralytics.com" target="_blank">
    <img src="https://storage.googleapis.com/ultralytics/logo/logoname1000.png" width="110"></a>
      <img src="https://user-images.githubusercontent.com/26833433/61591093-2b4d4480-abc2-11e9-8b46-d88eb1dabba1.jpg">
          <a href="https://itunes.apple.com/app/id1452689527" target="_blank">
    <img src="https://user-images.githubusercontent.com/26833433/50044365-9b22ac00-0082-11e9-862f-e77aee7aa7b0.png" width="180"></a>
    </td>
    <td>
      <img src="https://user-images.githubusercontent.com/26833433/61591100-55066b80-abc2-11e9-9647-52c0e045b288.jpg">
    </td>
  </tr>
</table>

**Comment by FeatheryW** [comment]
@glenn-jocher I'm using a MacOS and have just finished training my dataset. How would I go by detecting my custom object using my webcam and or iphone? I have followed the tutorial you linked above.

**Comment by glenn-jocher** [comment]
Run this to use webcam:

```bash
git clone https://github.com/ultralytics/yolov3
cd yolov3
python3 detect.py --webcam
```

**Comment by NOxDriver** [comment]
@glenn-jocher Is there anyway to then use a custom trained weights on ios?

**Comment by glenn-jocher** [comment]
@NOssendryver of course. That’s the entire point of what we do. See YOLOv5 to get started.

**Comment by NOxDriver** [comment]
@glenn-jocher thanks. I've got my trained weights, but I just can not get it to work.
What's the best way to get this done? 
Weights are in Darknet format. I can convert to Modelml, but nothing gets detected in the xcode project when I deploy to my iphone.

**Comment by glenn-jocher** [comment]
See YOLOV5 export tutorial 
https://github.com/ultralytics/yolov5#tutorials
